code/src/cljs/sixsq/nuvla/ui/deployment_dialog/events.cljs

```diff
@@ -258,7 +258,7 @@
         {:select "id, name, description, created, subtype, last-check, status"
          :filter (general-utils/join-and
                    (str "parent='" infra-id "'")
-                   (str "subtype='infrastructure-service-swarm'"))}
+                   (str "subtype='infrastructure-service-" (:subtype selected-infra-service) "'"))}
```

Comment:
existing kubernetes cluster use same subtype as infrastructure-service-swarm. So existing creds of kubernetes will not appear.

Reply:
This worked for me because I'm creating Swarm/K8s creds directly from job-engine right after COE is provisioned. Indeed, if someone manually adds cred on COE of type kubernetes, it gets registered as 'infrastructure-service-swarm' and it's not getting picked up by the query above. I didn't test that case. I'm working on a fix. Good catch!

Reply:
Addressed with this commit 9d7764d